src/components/ScrollShadows.tsx

```diff
@@ -63,6 +63,11 @@ export function ScrollShadows(props: ScrollShadowsProps) {
   const onResize = useCallback(() => scrollRef.current && updateScrollProps(scrollRef.current), []);
   useResizeObserver({ ref: scrollRef, onResize });
 
+  // Hack for sizing more complicated components that might not be fully drawn on initial paint (such as a GridTable)
+  useLayoutEffect(() => {
+    setTimeout(() => onResize(), 0);
```

Comment:
yeah I'm not loving the solution - this hack allows us to delay the execution until the current execution thread is completed by placing this callback into the async queue (but without additional delay) since we need the multiple paints to complete before actually trying to size. 
A cleaner option perhaps is to just use a normal `useEffect` which should be called _after_ render, but also not provide any dependencies so it's called after each render since the callback is cheap? 🤔 Doing more testing on that path...

Reply:
Fwiw my has been something in `useSetupColumnSizes` i.e. this guy:

```
  // Used to recalculate our columns sizes when columns change
  useEffect(
    () => {
      if (!calculateImmediately.current) {
        const width = resizeRef.current?.clientWidth;
        width && setTableAndColumnWidths(width);
      }
    },
    // TODO: validate this eslint-disable. It was automatically ignored as part of https://app.shortcut.com/homebound-team/story/40033/enable-react-hooks-exhaustive-deps-for-react-projects
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [columns, setTableAndColumnWidths],
  );
```

I also thought we had a "ruler" div in GridTable somewhere, but I'm not seeing it anymore/atm... :thinking: